bot/internal/bot/bot.go

```diff
@@ -62,6 +62,17 @@ type Client interface {
 	// CreateComment will leave a comment on an Issue or Pull Request.
 	CreateComment(ctx context.Context, organization string, repository string, number int, comment string) error
 
+	// ListStatefulComments finds comments on a given PR matching the provided tag
+	ListStatefulComments(ctx context.Context, organization string, repository string, number int, tag string) ([]github.Comment, error)
+
+	// CreateOrUpdateStatefulComment will create a comment on a Pull Request, or update a pre-existing comment if one is found with a matching tag.
+	// Returns the number of created or updated comments.
+	CreateOrUpdateStatefulComment(ctx context.Context, organization string, repository string, number int, comment string, tag string) (int, error)
+
+	// DeleteStatefulComment will delete all comments on a Pull Request found with a matching tag.
+	// Returns the number of deleted comments.
+	DeleteStatefulComment(ctx context.Context, organization string, repository string, number int, tag string) (int, error)
```

Comment:
"Stateful" is a little confusing. Maybe "ListCommentsByTag", "DeleteCommentByTag", etc. instead?